python/meep.i

```diff
@@ -199,16 +199,24 @@ static meep::vec py_kpoint_func_wrap(double freq, int mode, void *user_data) {
     return result;
 }
 
-void py_master_printf_wrap(const char *s) {
-    PySys_WriteStdout("%s", s);
-    static PyObject *py_stdout = NULL;
-    if (py_stdout == NULL) {
-      py_stdout = PySys_GetObject("stdout");
-    }
-    PyObject *result = PyObject_CallMethod(py_stdout, "flush", NULL);
+static void _do_master_printf(const char* stream_name, const char* text) {
+    PyObject* py_stream = PySys_GetObject((char*)stream_name); // arg is non-const on Python2
+    PyObject* result;
+
+    result = PyObject_CallMethod(py_stream, "write", "(s)", text);
+    Py_XDECREF(result);
+    result = PyObject_CallMethod(py_stream, "flush", NULL);
     Py_XDECREF(result);
```

Comment:
Can probably get rid of the `result` variable and just call `Py_XDECREF(PyObject_CallMethod(...));` directly.